easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -209,16 +210,26 @@ def configure_step(self, srcdir=None, builddir=None):
         options.append('-DCMAKE_FIND_USE_PACKAGE_REGISTRY=FALSE')
 
         if not self.cfg.get('allow_system_boost', False):
-            # don't pick up on system Boost if Boost is included as dependency
-            # - specify Boost location via -DBOOST_ROOT
-            # - instruct CMake to not search for Boost headers/libraries in other places
-            # - disable search for Boost CMake package configuration file
             boost_root = get_software_root('Boost')
             if boost_root:
+                boost_version = get_software_version('Boost')
+                # Check for older builds of Boost
+                cmake_files = glob.glob(os.path.join(boost_root, 'lib', 'cmake', 'boost_system-%s' % boost_version,
+                                                     'libboost_system-variant*-shared.cmake'))
+                if len(cmake_files) > 1 and 'libboost_system-variant-shared.cmake' in cmake_files:
+                    # disable search for Boost CMake package configuration file for older builds
```

Comment:
A bit more specific:
```suggestion
                    # disable search for Boost CMake package configuration files when conflicting variant configs are present (builds using the old EasyBlock)
```